Document missing Iceberg configuration properties

agrawalreetika opened the pull request:
## Description
Document missing Iceberg configuration properties

## Motivation and Context
Document missing Iceberg configuration properties

## Impact

## Test Plan

## Contributor checklist

- [ ] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [ ] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [ ] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [ ] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [ ] Adequate tests were added if applicable.
- [ ] CI passed.

## Release Notes

```
== NO RELEASE NOTE ==
```

steveburnett commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -168,58 +168,73 @@ Configuration Properties
+                                                        Thrift protocol. If multiple URIs are provided, the first
+                                                        URI is used by default, and the rest of the URIs are
+                                                        fallback metastores.
+                                                        Example: ``thrift://192.0.2.3:9083`` or
```
```suggestion
                                                        
                                                        Examples: ``thrift://192.0.2.3:9083`` or
```
Suggest adding a line above to format the `Example:` sentence in the build the way it appears in the code here.

steveburnett commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -168,58 +168,73 @@ Configuration Properties
+                                                        URI is used by default, and the rest of the URIs are
+                                                        fallback metastores.
+                                                        Example: ``thrift://192.0.2.3:9083`` or
+                                                        ``thrift://192.0.2.3:9083,thrift://192.0.2.4:9083``
```
```suggestion
                                                        ``thrift://192.0.2.3:9083,thrift://192.0.2.4:9083``.
```

steveburnett commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -168,58 +168,73 @@ Configuration Properties
+                                                        fallback metastores.
+                                                        Example: ``thrift://192.0.2.3:9083`` or
+                                                        ``thrift://192.0.2.3:9083,thrift://192.0.2.4:9083``
+                                                        This property is required if the
```
```suggestion
                                                        
                                                        This property is required if the
```
Suggest adding a line above to format the Example: sentence in the build the way it appears in the code here.

steveburnett commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -168,58 +168,73 @@ Configuration Properties
+                                                        This property is required if the
+                                                        ``iceberg.catalog.type`` is ``hive``. Otherwise, it will
+                                                        be ignored.
+
```
Consider moving the entry for ``iceberg.catalog.type`` to immediately follow the entry for ``hive.metastore.uri``, because the two are related. 

Of course, ``hive.metastore.uri`` is not the only one related to ``iceberg.catalog.type``. Maybe a good order for this table could be: 
``hive.metastore.uri``
``iceberg.catalog.type`` 
``iceberg.catalog.warehouse``
``iceberg.catalog.cached-catalog-num``
``iceberg.hadoop.config.resources``
then
``iceberg.file-format``
``iceberg.compression-codec``
then 
``iceberg.max-partitions-per-writer``
and the rest. 

Maybe move ``iceberg.file-format`` and ``iceberg.compression-codec`` to the beginning (if those are very important to the user), then the block of catalog type-connected properties, then ``iceberg.max-partitions-per-writer`` and the rest. 

My idea is to group the block of catalog type-connected properties together, and you would know best the relative importance of that block and the other properties in the table. Does this make sense?

steveburnett commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -168,58 +168,73 @@ Configuration Properties
+
+``iceberg.hadoop.config.resources``                     The path(s) for Hadoop configuration resources.
+                                                        ``Example: /etc/hadoop/conf/core-site.xml.`` This property
+                                                        is required if the iceberg.catalog.type is hadoop. Otherwise,
```
```suggestion
                                                        is required if the iceberg.catalog.type is ``hadoop``. Otherwise,
```

steveburnett commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -168,58 +168,73 @@ Configuration Properties
+                                                        ``LZ4``, and ``ZSTD``.
+
+``iceberg.catalog.type``                                The catalog type for Iceberg tables. The available values     ``hive``
+                                                        are ``hive``, ``hadoop``, and ``nessie`` corresponding to
```
```suggestion
                                                        are ``hive``, ``hadoop``, and ``nessie``.
```
Suggest ending the sentence here.

steveburnett commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -168,58 +168,73 @@ Configuration Properties
+
+``iceberg.catalog.type``                                The catalog type for Iceberg tables. The available values     ``hive``
+                                                        are ``hive``, ``hadoop``, and ``nessie`` corresponding to
+                                                        the catalogs in Iceberg.
```
```suggestion

```

steveburnett commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -168,58 +168,73 @@ Configuration Properties
+                                                        the catalogs in Iceberg.
+
+``iceberg.catalog.warehouse``                           The catalog warehouse root path for Iceberg tables.
+                                                        ``Example: hdfs://nn:8020/warehouse/path``
```
```suggestion
                                                        
                                                        Example: ``hdfs://nn:8020/warehouse/path``
```
Suggest add a line to left-justify the Example (as it seems you may have meant to), and reformat to remove "Example: " from the code formatting.

steveburnett commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -168,58 +168,73 @@ Configuration Properties
+                                                        Otherwise, it will be ignored.
+
+``iceberg.hadoop.config.resources``                     The path(s) for Hadoop configuration resources.
+                                                        ``Example: /etc/hadoop/conf/core-site.xml.`` This property
```
```suggestion
                                                        
                                                        Example: ``/etc/hadoop/conf/core-site.xml`` 
                                                        
                                                        This property
```

steveburnett commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -168,58 +168,73 @@ Configuration Properties
+
+``iceberg.hive-statistics-merge-strategy``              Determines how to merge statistics that are stored in the     ``NONE``
+                                                        Hive Metastore. The available values are ``NONE``,
+                                                        , ``USE_NULLS_FRACTION_AND_NDV``, ``USE_NULLS_FRACTIONS``
```
```suggestion
                                                        ``USE_NULLS_FRACTION_AND_NDV``, ``USE_NULLS_FRACTIONS``
```

steveburnett reviewed (changes requested):
Thanks, this is great! I had some nits of formatting and phrasing suggestions, and a suggestion to think about how best to order the properties in the table. Let me know what you think!

tdcmeehan reviewed (commented):
LGTM % @steveburnett comments

agrawalreetika commented:
Thanks @steveburnett for your review. Your comments look appropriate to me. I have updated the document accordingly.
Also, I have added a few more indentation fixes which were coming as a warning for the iceberg document. Please take a look at the updates.

steveburnett reviewed (approved):
LGTM! (docs)

Pull of updated branch, new local build, everything looks good. Thank you for the quick update!

tdcmeehan commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -168,58 +168,76 @@ Configuration Properties
+                                                        statistics. A value of 1 means a single record is equivalent
+                                                        to 1 millisecond of time difference.
+
+``iceberg.pushdown-filter-enabled``                     Experimental: Enable filter pushdown for Iceberg. This is     ``false``
```
Is this true?  I thought it worked with ORC files on Java workers.

agrawalreetika replied:
As per I am aware of this feature flag only make the Plan changes that are needed for Native as of now. There was limited support for iceberg table filter pushdown was added as part of https://github.com/prestodb/presto/pull/16964 which is available by default as it is, feature flag is not required.
@imjalpreet Are you aware of ORC files support?

hantangwangd replied:
As I know, this property is indeed only for Native worker now. I also wonder whether we could use the same filter pushdown logic for both java workers and native workers? I'm not familiar with native iceberg connector implementation, are there irreconcilable logics between the two?

tdcmeehan replied:
I'll merge for now, but I think it should work for ORC since this just delegates to Aria scan (select readers) for ORC.  We can clarify in a follow up PR.  CC: @yingsu00 @imjalpreet

imjalpreet replied:
In the current implementation, if this flag is set in the case of a Java worker, we fail the query and throw an exception that it is not supported irrespective of Parquet or ORC. The reason was we have not yet verified the current implementation when using ORC with Java Worker. The initial aim was to keep the Java code flow the same for now and use the new optimizer rule only in the case of a native worker. I can re-visit this once I wrap up and reach a stable state for Iceberg on Prestissimo. We can verify whether the current changes are sufficient for ORC with Java Worker or do we need any modifications. Based on that, we can change when we throw the exception.

hantangwangd reviewed (approved):
LGTM! Great work, thanks.
